### PR TITLE
feat: add tests, builds, and deployments to the app

### PR DESCRIPTION
**Pull request by Mohamad-Kamar**
## Description:
The pull request includes the code for setting up pipelines on CicleCI and deploying the code to Heroku.

## Instructions for QA:
Once this branch is pushed to master, the added config file is used by Circle CI to run Building and testing tasks on it.
To replicate the jobs that happen on this repo, copy and paste the config.yaml file inside the .circleci folder.

Heroku is also configured to automatically deploy any changes to the main branch into the staging pipeline.
The "prod-start" and "heroku-postbuild" are what make it possible.

### Screenshots:
The screenshot below shows that the build and test pipelines succeed for this branch as it has a config file, but fail for the main branch.
image

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ node_modules/
 /coverage/
 /lib/
 /dist/
+/.env
```
Comment (marounmelhem) [inline .gitignore:7]: Please add a `.env.example` file including all `.env` variables as empty (this file should be commited)

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+#
+# Define the jobs we want to run for this project
+jobs:
+  build:
+    docker:
+      - image: node:16
+
+    steps:
+      - checkout
+      - run: npm install
```
Comment (marounmelhem) [inline .circleci/config.yml]: use `yarn` for consistency, also applies to the below `test` job

**Review by marounmelhem — CHANGES_REQUESTED** [review]
Great work @Mohamad-Kamar! Also, awesome job on setting up a staging environment as well. 
A few small changes and this should be good to go.

File: .gitignore (hunk 1 above)
Comment (Mohamad-Kamar) [reply .gitignore:7]: The only environment variable currently available in the files is set in the app.js file, and it is being used for Circle CI configs (it is committed with the rest of the changes).
Will make sure to provide such examples in the follow up PRs.

File: .circleci/config.yml (hunk 2 above)
Comment (Mohamad-Kamar) [reply .circleci/config.yml]: Addressed in [this commit](https://github.com/Mohamad-Kamar/full-stack-app-ex-9/pull/3/commits/f7cf239fd2c0268d185423735da54a57dd085fa1)

File: .gitignore (hunk 1 above)
Comment (marounmelhem) [reply .gitignore:7]: Having only 1 variable in your `.env` file doesn't mean you shouldn't add an `.env.example`. Let's add one and please make sure not to resolve conversations yourself. The reviewer should resolve it if the change has been addressed and validated.

Comment (Mohamad-Kamar) [reply .gitignore:7]: What I meant is that there is no variable set in the my version of .env file itself. 
There is a file called app.json that sets an environment variable for hiroku to use. (I mentioned in my previous comment that it was for CircleCI, which was wrong).

I added an example env file, which will take effect on src/shared/config.js if implemented in an env file.

**Review by marounmelhem — APPROVED** [review]

**Pull request merged**